Classes/View/PreviewView.php

```diff
@@ -503,16 +503,25 @@ protected function getCookie()
      */
     protected function getRecords(PageLayoutView $view, array $row, $area)
     {
+        $pidCondition = sprintf(' AND pid = %d ', (integer) $row['pid']);
+        $workspaceCondition = '';
+        $workspaceId = $this->getActiveWorkspaceId();
+        if ($workspaceId) {
+            $workspaceCondition = sprintf(' AND (t3ver_wsid = %d OR t3ver_wsid = 0)', $workspaceId);
+            $pidCondition = sprintf(' AND (pid = %d OR (pid = -1 AND t3ver_wsid > 0)) ', (integer) $row['pid']);
+        }
```

Comment:
ok, it was just a thought, to keep this as "close" to core as possible for future proofing :)